src/DDTrace/Integrations/Drupal/DrupalIntegration.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace DDTrace\Integrations\Drupal;
+
+use DDTrace\GlobalTracer;
+use DDTrace\SpanData;
+use DDTrace\Integrations\Integration;
+use DDTrace\Tag;
+use DDTrace\Type;
+
+class DrupalIntegration extends Integration
+{
+    const NAME = 'drupal';
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    /**
+     * @return int
+     */
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return Integration::NOT_LOADED;
+        }
+
+        $rootScope = GlobalTracer::get()->getRootScope();
+        $rootSpan = null;
+
+        if (null === $rootScope || null === ($rootSpan = $rootScope->getSpan())) {
+            return Integration::NOT_LOADED;
+        }
+
+        $integration = $this;
+
+        \DDTrace\trace_method(
+            'Drupal\Core\DrupalKernel',
+            'handle',
+            function (SpanData $span, $args) use ($rootSpan) {
+                if (!isset($args[0])) {
+                    return;
+                }
+
+                $req = $args[0];
+
+                $route = $req->getPathInfo();
+                $rootSpan->overwriteOperationName($route);
+                $rootSpan->setTag(Tag::COMPONENT, 'drupal');
+
+                $span->name = 'drupal.kernel.handle';
+                $span->type = Type::WEB_SERVLET;
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Drupal\Core\DrupalKernel',
+            'handleException',
+            function (SpanData $span, $args, $retval) use ($rootSpan) {
+                $span->name = 'drupal.kernel.handleException';
+                $span->meta[Tag::COMPONENT] = 'drupal';
+                if (!(isset($retval) && \method_exists($retval, 'getStatusCode') && $retval->getStatusCode() < 500)) {
+                    $rootSpan->setError($args[0]);
```

Comment:
Is the exception content captured elsewhere?

Reply:
Removed the if here, my initial thinking was that the exception is set at `Bootstrap.php`, but it only sets the error message, not the exception.

So `setError` is always called when handling `drupal.kernel.handleException`